src/linear_algebra/alternating.lean

```diff
@@ -274,6 +288,33 @@ lemma map_congr_perm [fintype ι] (σ : equiv.perm ι) :
   g v = (equiv.perm.sign σ : ℤ) • g (v ∘ σ) :=
 by { rw [g.map_perm, smul_smul], simp }
 
+/-- If the arguments are linearly dependent then the result is `0`.
+
+TODO: Can the `division_ring` requirement be relaxed? -/
+lemma map_linear_dependent
+  {K : Type*} [division_ring K]
+  {M : Type*} [add_comm_group M] [semimodule K M]
+  {N : Type*} [add_comm_group N] [semimodule K N]
+  (f : alternating_map K M N ι) (v : ι → M)
+  (h : ¬linear_independent K v) :
+  f v = 0 :=
+begin
+  obtain ⟨s, g, h, i, hi, hz⟩ := linear_dependent_iff.mp h,
+  -- the part that uses `division_ring`
+  suffices : f (update v i (g i • v i)) = 0,
+  { apply (units.mk0 (g i) hz).smul_eq_zero.1,
+    rw units.coe_mk0,
+    rw [f.map_smul, function.update_eq_self i v] at this,
+    exact this, },
```

Comment:
```suggestion
  { rw [f.map_smul, function.update_eq_self, smul_eq_zero] at this,
    exact or.resolve_left this hz, },
```

still needs `division_ring K` though. Is `c * x = 0 ↔ c = 0 ∨ x = 0` true in any `ring`? I don't think so but I don't know for sure?

Reply:
Thanks for the suggestion, I think I found then forgot about `smul_eq_zero`.

My feeling is that probably the definition of `alternating_map` is subtly "wrong" for cases which aren't the reals, and that the eventual solution is to invert the reasoning here, and have `map_linear_dependent` be part of the definition from which `map_eq_zero_of_eq` can be derived from.

To be clear, I still think it would be good to get this proof in somewhere, even if it sits around in mathlib with stricter hypotheses than necessary for a while. I don't want to open the can of worms of changing the definition to work for more general cases, because those cases aren't interesting to me and I don't have the background to reason about them. Once the statement is in, someone can change the structure "axioms" later :)

Reply:
> Is `c * x = 0 ↔ c = 0 ∨ x = 0` true in any ring?

No, but its true in rings with `no_zero_divisors` I think. There's no equivalent class for `has_scalar` though, as far as I'm aware.

Reply:
You might be looking for *torsion-free modules*: https://stacks.math.columbia.edu/tag/0549
I'm quite sure we don't have those in mathlib yet.

Reply:
"looking for" is a bit strong - I only added this lemma because the definition I'd found of alternating maps said that there were three equivalent definitions (`map_eq_zero_of_eq`, `map_swap`, and the one in this PR), and I felt like for completeness there should be a lemma for each